variables.tf

```diff
@@ -0,0 +1,61 @@
+variable "aws_region" {
+  default     = "us-west-2"
+  type        = string
+  description = "Region to use for AWS services"
+}
+
+variable "lambda_file_name" {
+  default     = "app"
+  type        = string
+  description = "Name of the lambda function file"
+}
+
+variable "lambda_file_extention" {
+  default     = ".py"
+  type        = string
+  description = "The extention of the lambda file"
+  validation {
+    condition     = var.lambda_file_extention == ".py"
+    error_message = "the file has to be Python file"
+  }
+}
+
+variable "lambda_function_name" {
+  default     = "lambda_test"
+  type        = string
+  description = "The name of the lambda function from the Python file"
+}
+
+variable "lambda_run_time" {
+  default     = "python3.11"
+  type        = string
+  description = "The run time of the lambda function"
+  validation {
+    condition     = can(regex("python3.1\\d", var.lambda_run_time))
+    error_message = "Please ensure the run time has to be python3.10 or above"
+  }
+}
+
+variable "lambda_role_name" {
+  default     = "first-lambda-role"
+  type        = string
+  description = "The name of the ami role"
+}
+
+variable "lambda_permission_name" {
+  default     = "lambda-permissions"
+  type        = string
+  description = "The name of the ami role policy for accessing the lambda"
+}
+
+variable "file_type" {
+  default     = "zip"
```

Comment:
This and output path don't really need to be variables as they don't change

Reply:
And they won't change in a module either

Reply:
True true! thanks! I will remove them in the next PR. will merge this one and create a new PR